OSRMTextInstructions/OSRMTextInstructions.swift

```diff
@@ -263,23 +252,23 @@ public class OSRMInstructionFormatter: Formatter {
             }
             
             if scanner.scanString("}", into: nil) {
-                if let tokenType = TokenType(rawValue: token as! String) {
-                    var replacement: String
-                    switch tokenType {
-                    case .wayName: replacement = wayName
-                    case .destination: replacement = destination
-                    case .exit: replacement = exit
-                    case .rotaryName: replacement = rotaryName
-                    case .laneInstruction: replacement = laneInstruction ?? ""
-                    case .modifier: replacement = modifierConstant
-                    case .direction: replacement = directionFromDegree(degree: bearing)
-                    case .wayPoint: replacement = nthWaypoint
-                    }
-                    if tokenType == .wayName {
-                        result += replacement // already modified above
-                    } else {
-                        result += modifyValueByKey?(tokenType, replacement) ?? replacement
-                    }
+                let tokenType = OSRMTokenType(rawValue: token as! String)
+                var replacement: String
+                switch tokenType {
+                case OSRMTokenType.wayName: replacement = wayName
+                case OSRMTokenType.destination: replacement = destination
+                case OSRMTokenType.exit: replacement = exit
+                case OSRMTokenType.rotaryName: replacement = rotaryName
+                case OSRMTokenType.laneInstruction: replacement = laneInstruction ?? ""
+                case OSRMTokenType.modifier: replacement = modifierConstant
```

Comment:
NS_EXTENSIBLE_STRING_ENUM is a rather new feature that doesn't fully bridge to Swift apparently. I've seen this a few times where type cannot be inferred in switch statements even though it can be inferred in other places. I don't know the root cause.